addons/sale_project/models/sale_order.py

```diff
@@ -111,6 +122,31 @@ def action_view_project_ids(self):
             action['views'] = [(view_kanban_id, 'kanban'), (view_form_id, 'form')]
         return action
 
+    def action_view_milestone(self):
+        self.ensure_one()
+        default_project = self.project_ids and self.project_ids[0]
+        default_sale_line = default_project.sale_line_id or self.order_line and self.order_line[0]
+        return {
+            'type': 'ir.actions.act_window',
+            'name': _('Milestones'),
+            'domain': [('sale_line_id', 'in', self.order_line.ids)],
+            'res_model': 'project.milestone',
+            'views': [(self.env.ref('project.project_milestone_view_tree').id, 'tree')],
+            'view_mode': 'tree',
+            'help': _("""
+                <p class="o_view_nocontent_smiling_face">
+                    No milestones found. Let's create one!
+                </p><p>
+                    Track major progress points that must be reached to achieve success.
+                </p>
+            """),
+            'context': {
+                'default_project_id' : default_project.id,
+                'default_sale_line_id' : default_sale_line.id,
+                **self.env.context
```

Comment:
It could be better to add the keys/values in `self.env.context` before the new keys to avoid overriding those by the ones contained in `self.env.context`.
```suggestion
                **self.env.context,
                'default_project_id' : default_project.id,
                'default_sale_line_id' : default_sale_line.id,
```